code-challenges.test.js

```diff
@@ -1,46 +1,101 @@
-]
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
-// b) Create the function that makes the test pass.
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
-// a) Create a test with an expect statement using the variables provided.
+// Pseudo Code
+// Input: Array of mixed data
+// Output: An array of only the remainders of the numbers when divided by 3
+// The remainedersOnly function will take an array of mixed data as its parameters
+// Numbers will create another array that will only have numbers
+// By checking the typeOf of the data, it is easier to filter out the numbers
+// Remainders will create another array from the numbers mapped from the Numbers array
+// The map method will use each number and calculates the remainder of each number divided by 3 using the modulo operator
+// The returned array will only have the remainders from the numbers divided by 3
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+const remaindersOnly = (array) => {
+  const numbers = array.filter(element => typeof element === 'number')
+  const remainders = numbers.map(number => number % 3)
+  return remainders
+}
 
-// b) Create the function that makes the test pass.
+describe('remaindersOnly', () => {
+  it('should return an array of remainders of the numbers when divided by 3', () => {
+    const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+    const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+    expect(remaindersOnly(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(remaindersOnly(hodgepodge2)).toEqual([ 2, 1, -1 ])
+})
+})
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
-// a) Create a test with an expect statement using the variables provided.
+// Pseudo Code:
+// Input: array of numbers
+// Output: the sum of all the numbers cubed
+// The function cubedAdded will take in an array of numbers as its parameters
+// Sum will make an array from the numbers mapped from the given parameters and stores this info
+// Math.pow method will take all the numbers from the array and calculates its cube
+// After the map is done iterating over the array and all the numbers are cubed, the reduce method will then calculate the sum of all the numbers by applying a callback function to each number
+// Acc will represent the accumulator is the value that gets returned after each iteration and will change as the reduction progresses
+// Curr represents the current number it's processing in the array
+// The intial value will be set to 0
+// The sum variable will then return all the numbers cubed and the accumulated total
```

Comment:
Great step by step pseudo, really detailed what was happening in your code. You can also just return the whole thing instead of storing it in a variable.